contracts/HomoraIBSwap.sol

```diff
@@ -153,67 +142,51 @@ contract HomoraIBSwap is Governable {
         require(isIBToken[tokenIn], "token-in-not-supported");
         require(isIBToken[tokenOut], "token-out-not-supported");
 
-        SafeBox safeboxIn = SafeBox(tokenIn);
-        safeboxIn.transferFrom(msg.sender, address(this), amountIn);
-        safeboxIn.withdraw(amountIn);
-
         uint256 underlyingBalance;
-        address[] memory path = getPath(tokenIn, tokenOut);
         uint256 outputAmount;
+        address[] memory path = getPath(tokenIn, tokenOut);
 
-        if (tokenIn != IBETHV2) {
-            IERC20 underlying = IERC20(safeboxIn.uToken());
-            underlyingBalance = underlying.balanceOf(address(this));
+        SafeBox(tokenIn).transferFrom(msg.sender, address(this), amountIn);
+        SafeBox(tokenIn).withdraw(amountIn);
+
+        if (tokenIn == IBETHV2) {
```

Comment:
really minor gas opt:

use local variable to store IBETHV2 value (since it's accessed twice in this function).